ee/codegen/src/__test__/nodes/base-node.test.ts

```diff
@@ -0,0 +1,42 @@
+import { workflowContextFactory } from "src/__test__/helpers";
+import { templatingNodeFactory } from "src/__test__/helpers/node-data-factories";
+import { createNodeContext } from "src/context";
+import { TemplatingNodeContext } from "src/context/node-context/templating-node";
+import { NodeAttributeGenerationError } from "src/generators/errors";
+import { TemplatingNode } from "src/generators/nodes/templating-node";
+
+describe("BaseNode", () => {
+  describe("failures", () => {
+    it("should throw the expected error when a input references an invalid node", async () => {
+      const workflowContext = workflowContextFactory();
+
+      const templatingNodeData = templatingNodeFactory({
+        inputRules: [
+          {
+            type: "NODE_OUTPUT",
+            data: {
+              nodeId: "12345678-1234-5678-1234-567812345678",
+              outputId: "90abcdef-90ab-cdef-90ab-cdef90abcdef",
+            },
+          },
+        ],
+      });
+      const templatingNodeContext = (await createNodeContext({
+        workflowContext,
+        nodeData: templatingNodeData,
+      })) as TemplatingNodeContext;
+      workflowContext.addNodeContext(templatingNodeContext);
+
+      expect(() => {
+        new TemplatingNode({
+          workflowContext,
+          nodeContext: templatingNodeContext,
+        });
+      }).toThrow(
+        new NodeAttributeGenerationError(
+          "Failed to generate attribute 'TemplatingNode.inputs.text': Failed to find node with id '12345678-1234-5678-1234-567812345678'"
+        )
+      );
```

Comment:
In a future PR, instead of expecting this to throw, we change this to expect that the node has multiple warnings. thoughts on this API:
```typescript
const node = new TemplatingNode({
    workflowContext,
    nodeContext: templatingNodeContext,
});
const warnings = node.getWarnings();
expect(warnings).toBe([
    {
        "code": "NODE_ATTRIBUTE_GENERATION_ERROR",
        "message": "Failed to etc etc etc...",
    }
])
```

Reply:
I'm imagining that our CLI has a flag that's like `--strict=True`. If strict mode is enabled, we error instead of warn. If False, we do what we can. This applies to both codegen and serialization.

In this world, I'd want this to be called `getErrors()` because that's what they are, and then the flag determines what to do about those errors.